File: app/Models/Tenant/EmergencyContact.php
<?php

namespace App\Models\Tenant;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class EmergencyContact extends Model
{
    use SoftDeletes;

    protected $connection = 'society';

    protected $fillable = [
        'label',
        'phone',
        'type',
        'availability',
        'sort_order',
    ];
}

File: mobile/lib/features/emergency_contacts/data/emergency_contact.dart
class EmergencyContact {
  EmergencyContact({
    required this.id,
    required this.label,
    required this.phone,
    required this.type,
    this.availability,
  });

  factory EmergencyContact.fromJson(Map<String, dynamic> json) {
    return EmergencyContact(
      id: json['id'] as int,
      label: json['label'] as String,
      phone: json['phone'] as String,
      type: json['type'] as String,
      availability: json['availability'] as String?,
    );
  }

  final int id;
  final String label;
  final String phone;
  final String type;
  final String? availability;
}
